### DISCUSSION TITLE
Support for validation across parameters

### DISCUSSION THREAD

**Discussion by sazzer** [OAI/sig-moonwalk#31] [Ideas]
Right now in OpenAPI 3.x you have the ability to specify header, path and query parameters and to validate those parameters. Wha you *don't* have the ability to do is to have validation that spans across them.

For example, I might have a resource that supports two parameters:
* sortBy
* sortOrder

Where `sortOrder` is mandatory *if `sortBy` is specified*, but prohibited otherwise. Right now you can't do this, and instead you are forced to depend on free-text documentation of the fields.

Other things that would be useful here include, but not limited to:
* Restricting the values of one parameter based on the values of other parameters - as opposed to just the presence/absence of them.
* Changing the way parameters are treated based on HTTP headers.
* Mutually exclusive parameters - can have ether `a` or `b` but never both.

**Comment by handrews** [comment]
See also #20 and #24.

**Comment by handrews** [comment]
I'm going to close this in favor of #100, which got started by someone who did not realize this discussion already existed and now has a much longer set of comments.  Please feel free to add to that discussion!
